✨ Feat : Project - 후원하기 기능 구현

JeongO41 opened the pull request:
## PR 체크사항
PR이 다음 사항을 만족하는지 확인해주세요.

<!-- 
체크하려면 괄호 안에 "x"를 입력하세요. 
각 규칙은 Convention 문서에 있습니다.
PR 제목에 쓰는 prefix는 다음과 같습니다.
🚀 Release
🐛 Fix
✨ Feat
📝 Doc
♻️ Refactor
🔧 Chore
⏪️ Revert
🧪 Test
🎉 Init
-->

- [x] 커밋 제목 규칙
- [x] 커밋 메시지 작성 가이드라인
- [x] 라벨, 담당자, 리뷰어 지정


## PR 타입
어떤 유형의 PR인지 체크해주세요.

<!-- 체크하려면 괄호 안에 "x"를 입력하세요. -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:


## PR 설명
후원하기 기능 관련 로직 작성


## 작업사항
- SupportCreateRequestDto 작성
- ProjectController에  후원하기 관련 코드 작성(성현님 코드에 추가되어야 함)
- ProjectService에  후원하기 관련 코드 추가(성현님 코드에 추가되어야 함)
- SupportMapper 작성

close #30

JeongO41 committed:
[feat] Project - 후원하기 기능 구현

1. SupportCreateRequestDto 작성
2. ProjectController에  후원하기 관련 코드 작성(성현님 코드에 추가되어야 함)
3. ProjectService에  후원하기 관련 코드 추가(성현님 코드에 추가되어야 함)
4. SupportMapper 작성

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/project/mapper/SupportMapper.java:
```diff
@@ -0,0 +1,13 @@
+import com.team1.TBBCloneCoding.project.entity.Support;
+
+public class SupportMapper {
+    public Support toSupport(Member member, Project project, SupportCreateRequestDto supportCreateRequestDto) {
```
`@Builder`를 이용해서 Support객체를 만들어주세요,

JeongO41 committed:
[fix] Project 후원하기 코드 수정

1. SupportMapper - support객체 생성 시 builder 사용하도록 수정
2. SupportMapper - @Component 추가
3. ProjectController - @RestController 추가
4. SupportCreateRequestDto - @NoArgsConstructor 추가
